core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -585,6 +559,71 @@ public void addIndex(CharSequence columnName, int indexValueBlockSize) {
         LOG.info().$("ADDED index to '").utf8(columnName).$('[').$(ColumnType.nameOf(existingType)).$("]' to ").$(path).$();
     }
 
+    public void dropIndex(CharSequence columnName) {
+
+        checkDistressed();
+
+        final int columnIndex = getColumnIndexQuiet(metaMem, columnName, columnCount);
+        if (columnIndex == -1) {
+            throw CairoException.metadataValidation("Column does not exist", columnName);
+        }
+        if (!isColumnIndexed(metaMem, columnIndex)) {
+            // if a column is indexed, it is al so of type SYMBOL
+            throw CairoException.metadataValidation("Column is not indexed", columnName);
+        }
+        final int defaultIndexValueBlockSize = Numbers.ceilPow2(configuration.getIndexValueBlockSize());
+
+        try {
+            LOG.info().$("BEGIN DROP INDEX [txn=")
+                    .$(txWriter.getTxn())
+                    .$(", table=")
+                    .$(tableName)
+                    .$(", column=")
+                    .$(columnName)
+                    .I$();
+            // drop index
+            if (dropIndexOperator == null) {
+                dropIndexOperator = new DropIndexOperator(configuration, messageBus, this);
+            }
+            dropIndexOperator.executeDropIndex(tableName, columnName, columnIndex); // upserts column version in partitions
+            // swap meta commit
+            metaSwapIndex = copyMetadataAndSetIndexAttrs(columnIndex, META_FLAG_BIT_NOT_INDEXED, defaultIndexValueBlockSize);
+            swapMetaFile(columnName); // bumps structure version, this is in effect a commit
+
+            // refresh metadata
+            TableColumnMetadata columnMetadata = metadata.getColumnQuick(columnIndex);
+            columnMetadata.setIndexed(false);
+            columnMetadata.setIndexValueBlockCapacity(defaultIndexValueBlockSize);
+            // remove indexer
+            ColumnIndexer columnIndexer = indexers.getQuick(columnIndex);
```

Comment:
this is a bug. If indexer does exist, removing from `indexers` list will collapse it. E.g. it will have less items than number of columns (this is sparse list). If you had a test, where you have two indexed columns, lets say at positions 1 and 5. You remove index for column 1 and see if column 5 still works correctly.

There is another bug in so far as `denseIndexers` list is not updated. Both these lists contain same indexer instances. One sparse, one dense.

More tests needed.

Reply:
addressed